### PR TITLE
Loading animations

### PR DESCRIPTION
**Pull request by xLacrima**

### CODE REVIEW

File: frontend/src/app/ruleset-creator/ruleset-creator.component.ts [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
+        style({ transform: 'scale(0.9)', opacity: 0.2 }),
+        animate('.3s ease-out')
+      ])
+    ])
+  ]
 })
 export class RulesetCreatorComponent implements OnInit {
 
+
+  @HostBinding('[@slideInOut]')
```
Comment (Jaykju) [inline frontend/src/app/ruleset-creator/ruleset-creator.component.ts]: This line seems to cause problems and throws the following error:

> ERROR in `@HostBinding` parameter should be a property name, 'class.<name>', or 'attr.<name>'.

The initial animation does not disappear. I also tried a clean install.

Comment (Jaykju) [inline frontend/src/app/ruleset-creator/ruleset-creator.component.ts]: This throws an error even after a clean install.

Comment (xLacrima) [reply frontend/src/app/ruleset-creator/ruleset-creator.component.ts]: it shouldnt be there anyway i just forgot to remove it

File: frontend/src/app/rulesets-overview/rulesets-overview.component.html [hunk 2]
```diff
@@ -1,10 +1,19 @@
-<div class="ruleset-card new" [routerLink]="['new']">
-    <mat-icon>add</mat-icon>
-</div>
-<div *ngFor="let ruleset of rulesets" class="ruleset-card" [routerLink]="[ruleset.rulesetId, 'edit']">
-    <h2 class="title">{{ ruleset.name }}</h2>
-    <div class="content">
-        <p class="description" ovideEllipsis>{{ ruleset.description }}</p>
-        <div class="meta-data">{{ ruleset.createdAt | date }}, {{ ruleset.createdBy }}</div>
+
+<div *ngIf="(rulesets$ | async) as rulesets; else loading" class="rulesets-container" [@cardInAnimation]>
```
Comment (Jaykju) [inline frontend/src/app/rulesets-overview/rulesets-overview.component.html:2]: This produces an infinite loading spinner when there is an error while retrieving the rulesets. This should be handled by #98 in the future.

**Pull request merged**